internal/controllers/rest_controller.go

```diff
@@ -4,52 +4,110 @@ import (
 	"context"
 	"encoding/json"
 	"net/http"
+	"time"
 
 	"github.com/gin-gonic/gin"
 
+	"github.com/ydataai/authentication-service/internal/configurations"
+	authErrors "github.com/ydataai/authentication-service/internal/errors"
+	"github.com/ydataai/authentication-service/internal/handlers"
+	"github.com/ydataai/authentication-service/internal/models"
 	"github.com/ydataai/authentication-service/internal/services"
 	"github.com/ydataai/go-core/pkg/common/logging"
 	"github.com/ydataai/go-core/pkg/common/server"
 )
 
 // RESTController defines rest controller.
 type RESTController struct {
-	configuration RESTControllerConfiguration
-	oidcService   *services.OIDCService
+	configuration configurations.RESTControllerConfiguration
+	oidcService   services.OIDCServiceInterface
+	credentials   []handlers.CredentialsHandler
 	logger        logging.Logger
 }
 
 // NewRESTController initializes rest controller.
 func NewRESTController(
 	logger logging.Logger,
-	configuration RESTControllerConfiguration,
-	oidcService *services.OIDCService,
+	configuration configurations.RESTControllerConfiguration,
+	oidcService services.OIDCServiceInterface,
+	credentials []handlers.CredentialsHandler,
 ) RESTController {
 	return RESTController{
 		configuration: configuration,
 		oidcService:   oidcService,
+		credentials:   credentials,
 		logger:        logger,
 	}
 }
 
-// Boot initialize creating some routes.
+// Boot initializes creating some routes.
 func (rc RESTController) Boot(s *server.Server) {
-	s.Router.GET(rc.configuration.AuthServiceURL, gin.WrapF(rc.RedirectToOIDCProvider))
+	s.Router.GET(rc.configuration.AuthServiceURL, gin.WrapF(rc.AuthenticationSession))
 	s.Router.GET(rc.configuration.OIDCCallbackURL, gin.WrapF(rc.OIDCProviderCallback))
+	s.Router.GET(rc.configuration.LogoutURL, gin.WrapF(rc.Logout))
+}
+
+// AuthenticationSession is responsible for knowing if the user already has a valid credential or not.
+// If so, forward 200 OK + UserID Headers.
+// If not, begin OIDC Flow.
+func (rc RESTController) AuthenticationSession(w http.ResponseWriter, r *http.Request) {
```

Comment:
`AuthenticationSession` sounds a strange name. There is no session here, right, because by definition a session is a group of interactions between a user and an application that take place within a given timeframe.

Reply:
We can rename this function. Maybe `UserAuthentication`?
What do you think? some suggestions?

Reply:
agree.
i would suggest `CheckForAuthentication`, don't know. @renatoaguimaraes do you have another name suggestion?

Reply:
> agree. i would suggest `CheckForAuthentication`, don't know. @renatoaguimaraes do you have another name suggestion?

`CheckForAuthentication` seems a good name.

Reply:
I agree 👍